### ISSUE TITLE
Using a composite partition key

### ISSUE THREAD

**Issue by foxfl** [aws-amplify/amplify-cli#4905]
Hey! 

We are using amplify in some apps now and the work and progress you put into it is amazing, so first of all: thanks for this nice library. 

My question is: is it currently possible to generate a composite hash key with `@key` directive? 
From the docs here: https://docs.amplify.aws/cli/graphql-transformer/directives#key it does not seem so, as only the first field in the fields array will be used as the hash key. 
Is there a Plan to support this use case in the future?

Thanks :) 

**Which Category is your question related to?**
API

**Amplify CLI Version** 
4.24.2

**What AWS Services are you utilizing?**
DynamoDB, AppSync

**Comment by hanzhaodeng** [comment]
Yeah I am very confused about this too. Not sure why they don't support compositing partition key.
